src/backend/web/__mocks__/authentication.js

```diff
@@ -63,33 +63,36 @@ function checkUser(requireAdmin, redirect, req, res, next) {
   }
 }
 
-// We'll deal with admin via init(), just let this pass
-function administration() {
-  return function (req, res, next) {
-    next();
-  };
+/**
+ * If we're logged in, build and return the appropriate user type (User vs. Admin).
+ * If we're not logged in, return null.
+ */
+function createUser() {
+  if (!loggedInUser) {
+    return null;
+  }
+
+  if (loggedInUser.isAdmin) {
+    return new Admin(loggedInUser.name, loggedInUser.email, loggedInUser.id);
+  }
+  return new User(loggedInUser.name, loggedInUser.email, loggedInUser.id);
 }
 
 function protect(redirect) {
   return function (req, res, next) {
-    if (loggedInUser) {
-      req.user = new User(loggedInUser.name, loggedInUser.email, loggedInUser.id);
-    }
+    req.user = createUser();
     checkUser(false, redirect, req, res, next);
   };
 }
 
 function protectAdmin(redirect) {
   return function (req, res, next) {
-    if (loggedInUser) {
-      req.user = new Admin(loggedInUser.name, loggedInUser.email, loggedInUser.id);
-    }
+    req.user = createUser();
```

Comment:
Can we refactor protect and protectAdmin to be one function which the parameter checkUser could be checked within the function instead of two separate functions that are identical except for the false/true that is passed through?

Reply:
As well as in `web\authentication.js`

Reply:
(this is listed as a comment in case there was something else we didn't think of.)

Reply:
I reviewed this with @Grommers00

Reply:
The pattern I'm following with 2 functions managing a boolean is the difference between:

```js
protect();
protectAdmin();

// vs.

protect(true);
protect(false);
```

The former is self-documenting, while the latter requires you to go look up what the boolean is for.

We could have protect take an `options` object, so you could do:

```js
protect({ redirect: true, admin: true })
```

Would that be better?

Reply:
So, our thought process was could we grab it from the user object, (isAdmin = true) leads and then pass that through, are we wrong in thinking like that way?

Reply:
They are different.  `protect` is being defined to work for a user vs. an admin.  The bit about figuring out what you are happens before that.